mautic/mautic
# M3 add new cache bundle

Commits taken from https://github.com/mautic/mautic/pull/7522 (M2 pull request) and I added latest changes from our cloud version proved by time ;-) 

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | Y
| Related user documentation PR URL | 
| Related developer documentation PR URL | https://github.com/hluchas/mautic/blob/m3-add-new-cache-bundle/app/bundles/CacheBundle/README.md
| Issues addressed (#s or URLs) | 
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This bundle enables the developer to use common Symfony caching mechanism with tag-aware adapters. It implements FS, Memcache and redis adapters.

#### Steps to test this PR:
1. Apply PR + require composer install
2. Set up an adapter as described in documentation
3. Use the cache as described in documentation
4. Verify items are created (e.g. redis records)
5. Verify items expire
6. Clear cache and verify items are purged
